projects/storefrontapp-e2e-cypress/cypress/integration/vendor/cds/profile-tag.e2e-spec.ts

```diff
@@ -21,6 +21,62 @@ describe('Profile-tag component', () => {
 
     profileTagHelper.triggerLoaded();
   });
+  it('should send a CartChanged event on adding an item to cart', () => {
+    goToProductPage();
+    cy.get('cx-add-to-cart button.btn-primary').click();
+    cy.get('cx-added-to-cart-dialog .btn-primary');
+    cy.window().then(win => {
+      expect((<any>win).Y_TRACKING.eventLayer.length).to.equal(2);
+      expect((<any>win).Y_TRACKING.eventLayer[1]['name']).to.equal(
+        'CartSnapshot'
+      );
+      const cartPayload = JSON.stringify(
+        (<any>win).Y_TRACKING.eventLayer[1]['data']
+      );
+      expect(cartPayload).to.include('cart');
+      expect(cartPayload).to.include('code');
+    });
+  });
+  it('should send an additional CartChanged event on modifying the cart', () => {
+    goToProductPage();
+    cy.get('cx-add-to-cart button.btn-primary').click();
+    cy.get('cx-added-to-cart-dialog .btn-primary').click();
+    cy.get('cx-cart-item cx-item-counter')
+      .getByText('+')
+      .click();
+    cy.route('GET', '/rest/v2/electronics-spa/users/anonymous/carts/*').as(
```

Comment:
One question is, profile tag should be tested only with an anonymous user or we want to test something related to the user as well?

Moreover, using the cy.route, it will get picked up automatically at the beginning by cypress, therefore the position of where you place it doesn't matter. Ideally, it should be at the top level, where we will define the routes to be watched for.

^ conclusion, can you move the getRefreshedCart route to the first statement within it's respective `it`, so in this case line 47 to line 41 and same for the others

Reply:
1 - anon user and registered user? or anonymous suffice
2 - move the getRefreshedCart route as the first statement of its respective it declaration

Reply:
2.1 - or you can move it before the click for readability I guess